Plugins/Public/help_expanded/Main.cpp

```diff
@@ -0,0 +1,295 @@
+map<wstring, HELPSTRUCT> mapHelp; 
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Loading Settings
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void LoadSettings()
+{
+	mapHelp.clear();
+	returncode = DEFAULT_RETURNCODE;
+
+	char szCurDir[MAX_PATH];
+	GetCurrentDirectory(sizeof(szCurDir), szCurDir);
+	string scPluginCfgFile = string(szCurDir) + "\\flhook_plugins\\laz_help.cfg";
+	bPluginEnabled = IniGetB(scPluginCfgFile, "Config", "Enabled", true); // Allow things to be quickly disabled
+
+	HELPSTRUCT helpstruct;
+	list<INISECTIONVALUE> iniSection; // Create a new list to store our values
+	IniGetSection(scPluginCfgFile, "Commands", iniSection); // Which header will we use to fill the values
+
+	foreach(iniSection, INISECTIONVALUE, iter) // Loop through the list
+	{
+		if(iter->scKey == "help") // Imitate ini reader
+		{
+			wstring getValueString = stows(iter->scValue); // Get around the character limit glitch
+
+			int firstComma = getValueString.find(','); // Get index of first comma
+			int secondComma = getValueString.substr(firstComma + 1).find(','); // Get index of second comma
+			wstring wscParam = getValueString.substr(0, firstComma); // Our param for the command
+			helpstruct.wscTitle = Trim(getValueString.substr(firstComma + 1, secondComma)); // The title for our text box
+			helpstruct.wscContent = Trim(getValueString.substr(firstComma + secondComma + 2)); // The content of our text box. XML String.
+
+			mapHelp[wscParam] = helpstruct;
+		}
+	}
+	ConPrint(L"HELP MENUS: Loaded %u help menus\n", mapHelp.size());
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void HelpInfo(uint iClientID)
+{
+	PrintUserCmdText(iClientID, L"Below is a list of all possible helpmenu commands, but there is also /rules and /start for some basic information for new players.");
+	PrintUserCmdText(iClientID, L"The current help commands available are as follows:");
+	for (map<wstring, HELPSTRUCT>::iterator iter = mapHelp.begin(); iter != mapHelp.end(); iter++)
+	{
+		PrintUserCmdText(iClientID, L"helpmenu %s", iter->first.c_str());
+	}
+}
+
+bool ValidityCheck(uint iClientID)
+{
+	if (!bPluginEnabled) // Make sure that the plugin isn't disabled via the cfg
+	{
+		PrintUserCmdText(iClientID, L"The extended help menu is currently disabled.");
+		return false;
+	}
+
+	// Stops people locking themselves out by using the command when dead.
+	if (HkIsOnDeathMenu(iClientID))
+	{
+		PrintUserCmdText(iClientID, L"ERR: You must be alive to use this command."); // Dummy msg
+		return false;
+	}
+	return true;
+}
+
+void TextboxPopUp(uint iClientID, const wstring &wscTitle, const wstring &wscXML)
```

Comment:
Looks like this could just be:
HkChangeIDSString(iClientID, 500001, wscXML);
RulesOrStarted(iClientID, wscTitle, 500001);